packages/expo-ui/ios/ColorPickerView.swift

```diff
@@ -0,0 +1,41 @@
+// Copyright 2025-present 650 Industries. All rights reserved.
+
+import SwiftUI
+import ExpoModulesCore
+
+class ColorPickerProps: ExpoSwiftUI.ViewProps {
+  @Field var selection: UIColor = .clear
+  @Field var label: String?
+  @Field var supportsOpacity: Bool = true
+  var onValueChanged = EventDispatcher()
+}
+
+struct ColorPickerView: ExpoSwiftUI.View {
+  @EnvironmentObject var props: ColorPickerProps
+  @State var color: Color = .clear
+  @State private var previousHex: String = ""
+  var body: some View {
+    ColorPicker(props.label ?? "", selection: $color, supportsOpacity: props.supportsOpacity)
+      .onAppear {
+        color = Color(props.selection)
+        previousHex = colorToHex(props.selection)
+      }
+      .onChange(of: color) { newValue in
+        let newHex = colorToHex(UIColor(newValue))
+        if newHex != previousHex {
+          previousHex = newHex
+          let payload = ["value": newHex]
+          props.onValueChanged(payload)
+        }
+      }
+  }
```

Comment:
```suggestion
struct ColorPickerView: ExpoSwiftUI.View {
  @EnvironmentObject var props: ColorPickerProps
  @State private var previousHex: String = ""
  
  var selection: Binding<Color> { Binding(
    get: { props.selection },
    set: { props.selection = $0 }
  )}
  
  var body: some View {
    ColorPicker(props.label ?? "", selection: selection, supportsOpacity: props.supportsOpacity)
      .onAppear {
        previousHex = colorToHex(props.selection)
      }
      .onChange(of: selection.wrappedValue) { newValue in
        let newHex = colorToHex(UIColor(newValue))
        if newHex != previousHex {
          previousHex = newHex
          let payload = ["value": newHex]
          props.onValueChanged(payload)
        }
      }
  }
```
I haven't tried, but would a manual `Binding` work here?

Reply:
I just tried this out, but wasn't working as expected. Let me know if you see any issues with my updated impl based on your comments

Reply:
Cool, thanks for trying 👍 we can look into this some other time